htdocs/install/mysql/migration/19.0.0-20.0.0.sql

```diff
@@ -295,3 +295,21 @@ ALTER TABLE llx_projet ADD COLUMN ref_ext varchar(50) after ref;
 
 -- bookcal
 ALTER TABLE llx_bookcal_calendar ADD COLUMN entity integer DEFAULT 1 NOT NULL AFTER rowid;
+
+
+create table llx_societe_users
```

Comment:
@eldy the suggest process is bit different. Let us know what do you think about this

Version N+1 (aka version 20.X) :

1. Create the new table llx_societe_users and UI in Contact/Adress Tab will be available
2. the use of llx_societe_users will be implemented in document creation (like in Contact Roles)
3. make data consistency up and running

- every added user in this screen with "type_contact" internal role code like 'SALESREP%' will be added in llx_societe_commerciaux if not already exists
- data store in llx_societe_commerciaux will never be be deleted by this feature, the management of list of Sales Rep will remain with field "Sales Rep" on thidparty tab/card in llx_societe_commerciaux table
- if user remove sales rep in llx_societe_commerciaux, users concerned in llx_societe_users will be deleted whatever contact type in llx_societe_users.
- mean all query around user permission user->hasRight('societe', 'client', 'voir') (aka "see all thirdparties") won't be changed

Version N+2 (aka version 21.X) :

1. in Thirdparty module setup a new field "Default Contact Type for Sales Rep", defaulted with new contact type dictionaries entry "Sales Rep" (Module:Thirdparty/Source: Internal)
2. A SQL/PHP migration script will transpose all data from llx_societe_commerciaux to llx_societe_users with this new role
3. All SQL query around user permission user->hasRight('societe', 'client', 'voir') will be rewrite to use llx_societe_users with the "Default Contact Type for Sales Rep" module settings field
4. The table llx_societe_commerciaux should not be used anymore by dolibarr core query
5. remove import process societe_4 ("Third-parties Sales representatives (assign sales representatives/users to companies)")
6. Create new import process for llx_societe_users 
7. Hide field "Sales Rep" in thirdparty card

Version N+3 (aka version 22.X) :
8. Table llx_societe_commerciaux is deleted if still exist

Advantages :

- Manage SQL View between MarriaDB and PgSQL can be tricky : create view require different database user permissions)
- even in MarriaDB the standards backup process can be compromise
- let the actual way of doing until N+2 version, let time to previous dolibarr version users to update their data

Reply:
Duplicating data in 2 tables with same role looks dangerous for me. I would prefer a more simple step that using 2 tables that need data synchronized):
- Just add the new fields you need in the existing llx_societe_commerciaux.  The new column fk_c_type_contact must have a default value that is the "Default Contact Type for Sales Rep"
- Modify the code to use the "Default Contact Type for Sales Rep"
- Introduce the code that will use new types.

- Later, when we want, we can rename the table in a big bang manner into llx_societe_user.

Reply:
@eldy, I've change the data model in add new column in llx_societe_commerciaux as requested.

I have to manage foreign key on code from c_type_contact, because I can determine what will be the next ID created by migration script, and I would like to have a forein key

I also change the data dump to make llx_c_type_contact.rowid autoincrement as it is the case in table creation file

Also, not related to this PR but all DROP INDEX in migration aren't corerctly syntaxed for PGSQL....